pandas/tests/io/parser/test_parse_dates.py

```diff
@@ -1290,17 +1290,17 @@ def test_parse_dates_empty_string(all_parsers):
     tm.assert_frame_equal(result, expected)
 
 
+@pytest.mark.parametrize(
+    'reader',
+    [
+        'read_csv_check_warnings',
+        'read_table_check_warnings'
+    ]
+)
 def test_parse_dates_infer_datetime_format_warning(all_parsers):
-    # GH 49024
+    # GH 49024, 51017
     parser = all_parsers
-    data = "Date,test\n2012-01-01,1\n,2"
```

Comment:
This shouldn't go

Reply:
If I parametrise over parser.read_csv_check_warnings and parser.read_table_check_warnings, how will I be able to insert data?

From what I'm understanding, getattr(parser, reader) should extract the attribute values of 'read_csv_check_warnings' or 'read_table_check_warnings' from the parser object depending on what is defined under our parametrization.

I'm having trouble understanding how the interchangeability works and how I would perform the data insertion if the value of reader is interchanging.

Reply:
I was thinking something like
```
    data = "Date,test\n2012-01-01,1\n,2"
    getattr(parser, reader)(
        UserWarning,
        "The argument 'infer_datetime_format' is deprecated",
        StringIO(data),
        parse_dates=["Date"],
        infer_datetime_format=True,
    )
```
so that the first time you get `parser.read_csv_check_warnings`, and the second time `parser.read_table_check_warnings` - would that work?

Reply:
ah, thanks - looks like we can't use this `getattr` trick then, else the stacklevel count gets messed up

I think we may just need to copy-and-paste the test below, so one test has `parser.read_csv_check_warnings` and the other one has `parser.read_table_check_warnings`

Reply:
Actually, your original idea of parametrization does work. 

Pytest only fails when I am running it through my Docker virtual environment from my terminal app. However, it succeeds when I run it through the VSCode terminal.  I am running a Docker virtual environment through VSCode using the Dev Containers extension. 

I wonder how the two virtual environments are out of sync?